### PR TITLE
EPNG-4942: Fix wrongly referenced schemas upon creation of the OpenAPI Spec

### PR DESCRIPTION
**Pull request by nzolotova**
Create new models for each endpoint instead of reusing already generated

### CODE REVIEW

File: restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorMultiModelsTest.kt [hunk 1]
```diff
@@ -0,0 +1,1362 @@
...
+                    type = "STRING",
+                    optional = false
+                )
+            ),
+            responseFields = listOf(
+// 				FieldDescriptor(
+// 					path = "_id",
+// 					description = "The unique identifier of the order.",
+// 					type = "STRING"
+// 				),
```
Comment (jmewes) [inline restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorMultiModelsTest.kt]: No commented out code please.

Comment (nzolotova) [reply restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorMultiModelsTest.kt]: Fixed

File: restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorMultiModelsTest.kt [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+import java.io.File
+
+class OpenApi20GeneratorMultiModelsTest {
+
+    @Test
+    fun `should convert resource model with list in response to openapi`() {
+        val api = givenGetProductResourceModel()
+        val openapi = whenOpenApiObjectGenerated(api)
+
+        then(openapi.definitions).hasSize(6)
```
Comment (mduesterhoeft) [inline restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorMultiModelsTest.kt]: Can you please detail this assertion a bit more? which models do you expect?

**Comment by mduesterhoeft** [comment]
@nzolotova thanks for your contribution! I am really lacking a little bit of context here. What is the problem you are trying to solve? Maybe you even have examples (I do not have access to the JIRA issue)

File: restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorTest.kt [hunk 3]
```diff
@@ -168,6 +170,43 @@ class OpenApi20GeneratorTest {
...
+        val ordersSchema: Model  = givenModel(ordersFieldDescriptors)
+        val shopsSchema: Model  = givenModel(shopsFieldDescriptors)
+
+        val schemaNameAndSchemaMap = HashMap<String, Model>()
+
+        //when
+        whenExtractOrFindSchema(schemaNameAndSchemaMap, ordersSchema, shopsSchema)
+
+        //then
+        then(schemaNameAndSchemaMap.size).isEqualTo(2)
```
Comment (mduesterhoeft) [inline restdocs-api-spec-openapi-generator/src/test/kotlin/com/epages/restdocs/apispec/openapi2/OpenApi20GeneratorTest.kt]: Great! This test would even stay green if the equals method on ObjectProperty would work correctly and we would move back to the aggregation of equal schemas! 👍

**Comment by mduesterhoeft** [comment]
Released with `0.8.1`

**Pull request merged**